services/motion/builtin/builtin.go

```diff
@@ -122,18 +126,31 @@ func (ms *builtIn) Reconfigure(
 	ms.movementSensors = movementSensors
 	ms.slamServices = slamServices
 	ms.components = components
+	if ms.state != nil {
+		ms.state.Stop()
+	}
+	ms.state = state.NewState(ms.logger)
```

Comment:
If this is the first time viam-server is configuring the MotionService, it will create a new state & add it to the builtIn struct. 

Every subsequent reconfigure we will stop any executions from before the reconfigure & create a new state.

Reply:
https://viam.atlassian.net/browse/RSDK-5158

Reply:
If I understand correctly, reconfiguration will clear all plan history, which is not what we want.

Reply:
That is indeed what this does. I thought that Eliot wanted the motion service to behave in the same manner as `AlwaysRebuild` which would have blown away the plan history as well. We can't use `AlwaysRebuild` as the motion service has weak dependencies which results in it always being in a reconfigure loop. see: https://github.com/viamrobotics/rdk/pull/2946 where we tried to change the motion service to always rebuild & had to revert it as it broke the RDK.

This is the comment that indicated to me that deleting plan state history on reconfiguration is the desired behavior: https://viam.atlassian.net/browse/RSDK-5158?focusedCommentId=22394

@raybjork @tessavitabile can y'all please let me know what the behavior should be?

Reply:
Yes Nick, I am also of the understanding that this is the behavior we agreed on

Reply:
I think this is fine. It's a little at odds with what we scoped for ListPlanStatuses: https://github.com/viamrobotics/api/blob/8fc476f421f79d8ea1a8efdffdb670c3559e8b6b/proto/viam/service/motion/v1/motion.proto#L56-L58. But I think it makes sense, since either way navigation will need to retry a waypoint if the status for it is unavailable. I think we should just update that comment to say that statuses will be clear on reinitialization or reconfiguration.